docs/book/src/topics/ignition-support.md

```diff
@@ -0,0 +1,127 @@
+# Ignition support
+
+The default configuration engine for bootstrapping workload cluster machines is [cloud-init][cloud-init].
+**Ignition** is an alternative engine used by Linux distributions such as [Flatcar Container Linux][flatcar]
+and [Fedora CoreOS][fedora-coreos] and therefore should be used when choosing an Ignition-based distribution as
+the underlying OS for workload clusters.
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This initial implementation uses Ignition **v2** and was tested with **Flatcar Container Linux** only.
+Future releases are expected to add Ignition **v3** support and cover more Linux distributions.
+
+</aside>
+
+This document explains how Ignition support works.
+
+For more generic information, see [Cluster API documentation on Ignition Bootstrap configuration][cabpk].
+
+## Overview
+
+By default machine controller stores EC2 instance user data using SSM to store it encrypted, which underneath
+use multi part mime types, which are [unlikely to be supported](https://github.com/coreos/ignition/issues/1072)
+by Ignition.
+
+EC2 user data is also limited to 64 KB, which is often not enough to provision Kubernetes controlplane because
+of the size of required certificates and configuration files.
+
+To address those limitations CAPA can create and use S3 Bucket to store encrypted user data, which will be then
+pulled by the instances during provisioning.
+
+## IAM Permissions
+
+To manage S3 Buckets and objects inside them, CAPA controllers require additional IAM permissions.
+
+If you use `clusterawsadm` for managing the IAM roles, you can use the configuration below to create S3-related
+IAM permissions.
+
+``` yaml
+apiVersion: bootstrap.aws.infrastructure.cluster.x-k8s.io/v1beta1
+kind: AWSIAMConfiguration
+spec:
+  s3Buckets:
+    enable: true
+```
+
+See [Using clusterawsadm to fulfill prerequisites](./using-clusterawsadm-to-fulfill-prerequisites.md) for more
+details.
+
+## Enabling BOOTSTRAP_FORMAT_IGNITION feature gate
+
+When deploying CAPA using `clusterctl`, make sure you set `BOOTSTRAP_FORMAT_IGNITION=true` and
+`EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION=true `environment variables to enable experimental Ignition bootstrap
+support.
+
+``` sh
+# Enable the feature gates controlling Ignition bootstrap.
+export EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION=true # Used by the kubeadm bootstrap provider.
+export BOOTSTRAP_FORMAT_IGNITION=true # Used by the AWS provider.
```

Comment:
Sorry for not asking this sooner. Do you think it would be better UX if we used the same env var for controlling our feature gate? I'm undecided

Reply:
Ah, I see that you also use `EXP_` prefix for some feature gates. Perhaps I can fix this. But for the `_KUBEADM_` part, I wouldn't really add it, as in this provider support is potentially wider than just for KABPK. So I'd change it to `EXP_BOOTSTRAP_FORMAT_IGNITION`. What do you think?

Reply:
On actually waking my brain up after a few cups of tea, using the same env var as CABPK does isn't a good idea, ignore me :smiley: 

With the `EXP_` prefix i would only use this if this feature is experimental. Correct me if i'm wrong but i don't think we are treating this as experimental (i.e. use at your own peril).

When we graduated EKS out of experimental we dropped the `EXP_` prefix and went with a `CAPA_` prefix. As a provider we haven't been consistent with the naming of our environment variables in relation to feature flags.  As this is a new feature what do you think to using a `CAPA_` prefix or not?

Reply:
Hmm, so to me it seems any kind of feature behind a feature gate is sort of experimental, but I can't say for sure. With CAPI and CABPK, Ignition support is right now treated as experimental as far as I know (due to fact that the bootstrap format should be decoupled from the bootstrap provider etc.) and I was under the impression (or maybe there was some review comment suggesting it here) that it should be treated similar in CAPA.

`CAPA_` prefix also sounds reasonable to me. Please decide as a maintainers and I will align accordingly.

Reply:
If its experimental upstream in CAPI then we will need to treat it the same (like we do for machine pools).  So can we change it to a `EXP_` prefix?

Reply:
Sure. I found the comment I had in mind also: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1979#issuecomment-929516769